File: apps/telegram/ui/cards.py
"""Card-style messages — mirrors telegram/cards.ts."""

from __future__ import annotations


def _bold(text: str) -> str:
    return f"*{text}*"


def main_menu_message() -> str:
    return (
        "*pmxtrader*\n\n"
        "Choose a section below. Live trades always need Confirm — never one-tap execute.\n\n"
        "Paste a Kalshi or Polymarket URL · /menu anytime"
    )


def market_card(
    *,
    title: str,
    venue: str,
    outcome: str,
    price: str,
    volume: str = "",
    url: str = "",
    updated_at: str = "",
) -> str:
    lines = [
        _bold("Market"),
        title,
        f"Venue: {venue}",
        f"Outcome: {outcome}",
        f"Price: {price}",
    ]
    if volume:
        lines.append(f"Volume: {volume}")
    if updated_at:
        lines.append(f"Updated: {updated_at}")
    if url:
        lines.append(f"\n{url}")
    return "\n".join(lines)


def trade_confirm_card(
    *,
    venue: str,
    market: str,
    outcome: str,
    side: str,
    size: str,
    est_cost: str,
    command: str,
) -> str:
    return (
        f"{_bold('Trade confirmation')}\n"
        f"Venue: {venue}\n"
        f"Market: {market}\n"
        f"Outcome: {outcome}\n"
        f"Side: {side} · Size: {size}\n"
        f"Est. cost: {est_cost}\n\n"
        f"{_bold('Command')}\n"
        f"`{command}`\n\n"
        "Tap Confirm — never executes from a single button alone."
    )


def agent_status_card(
    *,
    mode: str,
    provider: str,
    kill_switch: str,
    read_only: str,
    session_id: str = "",
    last_action: str = "",
) -> str:
    lines = [
        _bold("Agent status"),
        f"Mode: {mode}",
        f"Provider: {provider}",
        f"Kill switch: {kill_switch}",
        f"Read-only: {read_only}",
    ]
    if session_id:
        lines.append(f"Session: {session_id[:8]}…")
    if last_action:
        lines.append(f"Last: {last_action}")
    return "\n".join(lines)


def alert_card(
    *,
    title: str,
    condition: str,
    market: str,
    status: str,
    triggered_at: str = "",
) -> str:
    lines = [
        _bold("Alert"),
        title,
        f"Condition: {condition}",
        f"Market: {market}",
        f"Status: {status}",
    ]
    if triggered_at:
        lines.append(f"Triggered: {triggered_at}")
    return "\n".join(lines)


def portfolio_card(
    *,
    kalshi_balance: str,
    poly_balance: str,
    open_positions: int,
    day_pnl: str = "",
    risk_note: str = "",
) -> str:
    lines = [
        _bold("Portfolio"),
        f"Kalshi: {kalshi_balance}",
        f"Polymarket US: {poly_balance}",
        f"Open positions: {open_positions}",
    ]
    if day_pnl:
        lines.append(f"Day P&L: {day_pnl}")
    if risk_note:
        lines.append(f"\n{risk_note}")
    return "\n".join(lines)


def loading_message(context: str) -> str:
    return f"Checking {context}…"


def error_message(context: str, detail: str) -> str:
    return f"{_bold('Error')} — {context}\n\n{detail}"


def help_card() -> str:
    return (
        f"{_bold('pmxtrader help')}\n"
        "Paste a Kalshi or Polymarket URL for Scout research.\n"
        "Use /menu anytime for inline buttons.\n\n"
        "Safety:\n"
        "• Live trades need Confirm + Execute (two taps)\n"
        "• Briefs need approved: true before Trader\n"
        "• ./pmx activate-live before real orders"
    )

File: telegram-ui/cards.ts
/**
 * Card-style Telegram messages (Markdown).
 */

export interface MarketCardData {
  title: string;
  venue: string;
  outcome: string;
  price: string;
  volume?: string;
  url?: string;
  updatedAt?: string;
}

export interface TradeConfirmCardData {
  venue: string;
  market: string;
  outcome: string;
  side: string;
  size: string;
  estCost: string;
  command: string;
}

export interface AgentStatusCardData {
  mode: string;
  provider: string;
  sessionId?: string;
  killSwitch: string;
  readOnly: string;
  lastAction?: string;
}

export interface AlertCardData {
  title: string;
  condition: string;
  market: string;
  status: string;
  triggeredAt?: string;
}

export interface PortfolioCardData {
  kalshiBalance: string;
  polyBalance: string;
  openPositions: number;
  dayPnl?: string;
  riskNote?: string;
}

function bold(text: string): string {
  return `*${text}*`;
}

export function marketCard(data: MarketCardData): string {
  const lines = [
    bold("Market"),
    data.title,
    `Venue: ${data.venue}`,
    `Outcome: ${data.outcome}`,
    `Price: ${data.price}`,
  ];
  if (data.volume) lines.push(`Volume: ${data.volume}`);
  if (data.updatedAt) lines.push(`Updated: ${data.updatedAt}`);
  if (data.url) lines.push(`\n${data.url}`);
  return lines.join("\n");
}

export function tradeConfirmCard(data: TradeConfirmCardData): string {
  return [
    bold("Trade confirmation"),
    `Venue: ${data.venue}`,
    `Market: ${data.market}`,
    `Outcome: ${data.outcome}`,
    `Side: ${data.side} · Size: ${data.size}`,
    `Est. cost: ${data.estCost}`,
    "",
    bold("Command"),
    `\`${data.command}\``,
    "",
    "Tap Confirm — never executes from a single button alone.",
  ].join("\n");
}

export function agentStatusCard(data: AgentStatusCardData): string {
  const lines = [
    bold("Agent status"),
    `Mode: ${data.mode}`,
    `Provider: ${data.provider}`,
    `Kill switch: ${data.killSwitch}`,
    `Read-only: ${data.readOnly}`,
  ];
  if (data.sessionId) lines.push(`Session: ${data.sessionId.slice(0, 8)}…`);
  if (data.lastAction) lines.push(`Last: ${data.lastAction}`);
  return lines.join("\n");
}

export function alertCard(data: AlertCardData): string {
  const lines = [
    bold("Alert"),
    data.title,
    `Condition: ${data.condition}`,
    `Market: ${data.market}`,
    `Status: ${data.status}`,
  ];
  if (data.triggeredAt) lines.push(`Triggered: ${data.triggeredAt}`);
  return lines.join("\n");
}

export function portfolioCard(data: PortfolioCardData): string {
  const lines = [
    bold("Portfolio"),
    `Kalshi: ${data.kalshiBalance}`,
    `Polymarket US: ${data.polyBalance}`,
    `Open positions: ${data.openPositions}`,
  ];
  if (data.dayPnl) lines.push(`Day P&L: ${data.dayPnl}`);
  if (data.riskNote) lines.push(`\n${data.riskNote}`);
  return lines.join("\n");
}

export function loadingMessage(context: string): string {
  return `Checking ${context}…`;
}

export function errorMessage(context: string, detail: string): string {
  return `${bold("Error")} — ${context}\n\n${detail}`;
}

export function helpCard(): string {
  return [
    bold("pmxtrader help"),
    "Paste a Kalshi or Polymarket URL for Scout research.",
    "Use /menu anytime for inline buttons.",
    "",
    "Safety:",
    "• Live trades need Confirm + Execute (two taps)",
    "• Briefs need approved: true before Trader",
    "• ./pmx activate-live before real orders",
  ].join("\n");
}
